pkg/cmd/pulumi/state/state_taint.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package state
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/pulumi/pulumi/pkg/v3/backend/display"
+	"github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/backend"
+	"github.com/pulumi/pulumi/pkg/v3/resource/deploy"
+	pkgWorkspace "github.com/pulumi/pulumi/pkg/v3/workspace"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/env"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+
+	"github.com/spf13/cobra"
+)
+
+func newStateTaintCommand() *cobra.Command {
+	var stack string
+	var yes bool
+
+	cmd := &cobra.Command{
+		Use:   "taint [resource URN...]",
+		Short: "Taint one or more resources in the stack's state",
+		Long: `Taint one or more resources in the stack's state.
+
+This has the effect of ensuring the resources are destroyed and recreated upon the next ` + "`pulumi up`" + `.
```

Comment:
Would `pulumi refresh` also replace them?  I *think* it would, but not sure.  If so we should probably mention it.